NumSet/NumSet.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+
+namespace Assingment4.Tests
+{
+    [TestClass]
+    public class NumSetTests
+    {
+        [TestMethod]
+        public void Constructor_PassesParamsArray_AssignsParamsArrayToHashSet()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            HashSet<int> temp = new HashSet<int>(intArray);
+
+            //Act
+            NumSet numSet = new NumSet(intArray);
+
+            //Assert
+            Assert.IsNotNull(numSet.IntSet);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Constructor_PassedNullArgs_ThrowsException()
+        {
+            var numSet = new NumSet(null);
+        }
+
+        [TestMethod]
+        public void ReturnArray_TakesLocalHashSet_ReturnsAnIntArray()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            NumSet numSet = new NumSet(intArray);
+
+            //Act
+            int[] returnArray = numSet.ReturnArray();
+
+            //Assert
+            CollectionAssert.AreEqual(intArray, returnArray);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ReturnArray_HashSetHasNotBeenSetupByConstructor_ThrowsException()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            NumSet numSet = new NumSet(intArray);
+            numSet.IntSet = null;
+
+            //Act
+            numSet.ReturnArray();
+
+            //Assert
+
+        }
+
+        [TestMethod]
+        public void Equals_PassedTwoDifferentNumSet_ReturnsFalse()
+        {
+            //Assign
+            int[] intArray1 = new int[] { 1, 2, 3 };
+            int[] intArray2 = new int[] { 1, 2, 4 };
```

Comment:
```suggestion
            int[] intArray2 = new int[] { 1, 2, 3 };
```
If you changed your second array to match your first array as I've done above this test method will still pass, which is not what you want because now the two NumSets are equal to eachother based on the assignments criteria.